pkg/skaffold/actions/runner.go

```diff
@@ -72,8 +77,10 @@ func (r Runner) Exec(ctx context.Context, out io.Writer, allbuilds []graph.Artif
 
 	a := acs[0]
 
-	defer r.cleanup(context.TODO(), out, []Task{a}, []ExecEnv{execEnv})
-	return a.Exec(ctx, out)
+	err = a.Exec(ctx, out)
+	log.Entry(ctx).Debugf("Finished execution for %v", a.name)
+	r.cleanup(context.TODO(), out, []Task{a}, []ExecEnv{execEnv})
```

Comment:
nit: should use the current `ctx` instead of `context.TODO()`?

Reply:
You are right, fixed. Thanks!